src/include/migraphx/program.hpp

```diff
@@ -126,22 +74,21 @@ struct program
 
     void debug_print() const;
     void debug_print(instruction_ref ins) const;
-    void debug_print(const std::vector<instruction_ref>& inss) const;
     void print_graph(std::ostream& os, bool brief = false) const;
     void print_cpp(std::ostream& os) const;
 
     void dry_run(parameter_map params) const;
 
-    void annotate(std::ostream& os, std::function<void(instruction_ref)> a) const;
+    void annotate(std::ostream& os, const std::function<void(instruction_ref)>& a) const;
```

Comment:
This function in the program should probably pass the current module to the function(via `std::function<void(module* m, instruction_ref)>`.

Reply:
This function calls the module method annotate(), which already have the module information.